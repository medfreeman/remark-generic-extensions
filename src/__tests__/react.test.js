import { transformToReact } from "./helpers/transform";

describe("react inline extensions", () => {
  test("should add an extension", () =>
    expect(transformToReact("!Icon")).toMatchSnapshot());

  test("should add an extension with an id", () =>
    expect(transformToReact("!Icon{ #my-id }")).toMatchSnapshot());

  test("should add an extension with a class", () =>
    expect(transformToReact("!Icon{ .my-class }")).toMatchSnapshot());

  test("should add an extension with an attribute", () =>
    expect(transformToReact("!Icon{ attr=my-attr }")).toMatchSnapshot());

  test("should add an extension with a quoted attribute", () =>
    expect(transformToReact(`!Icon{ attr="my attribute" }`)).toMatchSnapshot());

  test("should add an extension with a lone attribute", () =>
    expect(transformToReact(`!Icon{ attr }`)).toMatchSnapshot());

  test("should add an extension with a lone attribute with a default value", () =>
    expect(
      transformToReact(`!Icon{ attr }`, {
        elements: {
          Icon: {
            propsDefaultValues: {
              attr: "default"
            }
          }
        }
      })
    ).toMatchSnapshot());

  test("should add an extension with a lone attribute with a default boolean value", () =>
    expect(
      transformToReact(`!Icon{ highlight }`, {
        elements: {
          Icon: {
            propsDefaultValues: {
              highlight: true
            }
          }
        }
      })
    ).toMatchSnapshot());

  test("should add an extension with all types of parameters", () =>
    expect(
      transformToReact(
        `!Icon{ #my-id .my-class attr attr2=my-attr attr3="my attribute" }`,
        {
          elements: {
            Icon: {
              propsDefaultValues: {
                attr: "default"
              }
            }
          }
        }
      )
    ).toMatchSnapshot());

  test("should add an extension with a boolean property", () =>
    expect(
      transformToReact("!Icon", {
        elements: {
          Icon: {
            html: {
              properties: {
                highlight: true
              }
            }
          }
        }
      })
    ).toMatchSnapshot());

  test("should add an extension with a mapped content", () =>
    expect(
      transformToReact("!Icon[my-tooltip]", {
        elements: {
          Icon: {
            html: {
              properties: {
                tooltip: "::content::"
              }
            }
          }
        }
      })
    ).toMatchSnapshot());

  test("should add an extension with a mapped argument", () =>
    expect(
      transformToReact("!Icon(my-icon)", {
        elements: {
          Icon: {
            html: {
              properties: {
                icon: "::argument::"
              }
            }
          }
        }
      })
    ).toMatchSnapshot());

  test("should work with other markdown", () =>
    expect(
      transformToReact(
        `# heading

!Icon`
      )
    ).toMatchSnapshot());
});

describe("react block extensions", () => {
  test("should add an extension", () =>
    expect(
      transformToReact(
        `Icon:
:::
:::`
      )
    ).toMatchSnapshot());

  test("should add an extension with an id", () =>
    expect(
      transformToReact(
        `Icon:
:::
:::
{ #my-id }`
      )
    ).toMatchSnapshot());

  test("should add an extension with a class", () =>
    expect(
      transformToReact(
        `Icon:
:::
:::
{ .my-class }`
      )
    ).toMatchSnapshot());

  test("should add an extension with an attribute", () =>
    expect(
      transformToReact(
        `Icon:
:::
:::
{ attr=my-attr }`
      )
    ).toMatchSnapshot());

  test("should add an extension with a quoted attribute", () =>
    expect(
      transformToReact(
        `Icon:
:::
:::
{ attr="my attribute" }`
      )
    ).toMatchSnapshot());

  test("should add an extension with a lone attribute", () =>
    expect(
      transformToReact(
        `Icon:
:::
:::
{ attr }`
      )
    ).toMatchSnapshot());

  test("should add an extension with a lone attribute with a default value", () =>
    expect(
      transformToReact(
        `Icon:
:::
:::
{ attr }`,
        {
          elements: {
            Icon: {
              propsDefaultValues: {
                attr: "default"
              }
            }
          }
        }
      )
    ).toMatchSnapshot());

  test("should add an extension with a lone attribute with a default boolean value", () =>
    expect(
      transformToReact(
        `Icon:
:::
:::
{ highlight }`,
        {
          elements: {
            Icon: {
              propsDefaultValues: {
                highlight: true
              }
            }
          }
        }
      )
    ).toMatchSnapshot());

  test("should add an extension with all types of parameters", () =>
    expect(
      transformToReact(
        `Icon:
:::
:::
{ #my-id .my-class attr attr2=my-attr attr3="my attribute" }`,
        {
          elements: {
            Icon: {
              propsDefaultValues: {
                attr: "default"
              }
            }
          }
        }
      )
    ).toMatchSnapshot());

  test("should add an extension with a boolean property", () =>
    expect(
      transformToReact(
        `Icon:
:::
:::`,
        {
          elements: {
            Icon: {
              html: {
                properties: {
                  highlight: true
                }
              }
            }
          }
        }
      )
    ).toMatchSnapshot());

  test("should add an extension with a mapped content", () =>
    expect(
      transformToReact(
        `Icon:
:::
my-tooltip
:::`,
        {
          elements: {
            Icon: {
              html: {
                properties: {
                  tooltip: "::content::"
                }
              }
            }
          }
        }
      )
    ).toMatchSnapshot());

  test("should add an extension with a mapped argument", () =>
    expect(
      transformToReact(
        `Icon: my-icon
:::
:::`,
        {
          elements: {
            Icon: {
              html: {
                properties: {
                  icon: "::argument::"
                }
              }
            }
          }
        }
      )
    ).toMatchSnapshot());

  test("should work with other markdown", () =>
    expect(
      transformToReact(
        `# heading

Icon:
:::
:::`
      )
    ).toMatchSnapshot());
});
