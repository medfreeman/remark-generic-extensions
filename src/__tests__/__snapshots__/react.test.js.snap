// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react block extensions should add an extension 1`] = `
<icon>
  
</icon>
`;

exports[`react block extensions should add an extension with a boolean property 1`] = `
<icon
  highlight={true}
>
  
</icon>
`;

exports[`react block extensions should add an extension with a class 1`] = `
<icon
  className="my-class"
>
  
</icon>
`;

exports[`react block extensions should add an extension with a lone attribute 1`] = `
<icon
  attr=""
>
  
</icon>
`;

exports[`react block extensions should add an extension with a lone attribute with a default boolean value 1`] = `
<icon
  highlight={true}
>
  
</icon>
`;

exports[`react block extensions should add an extension with a lone attribute with a default value 1`] = `
<icon
  attr="default"
>
  
</icon>
`;

exports[`react block extensions should add an extension with a mapped argument 1`] = `
<icon
  icon="my-icon"
>
  
</icon>
`;

exports[`react block extensions should add an extension with a mapped content 1`] = `
<icon
  tooltip="my-tooltip"
/>
`;

exports[`react block extensions should add an extension with a quoted attribute 1`] = `
<icon
  attr="my attribute"
>
  
</icon>
`;

exports[`react block extensions should add an extension with all types of parameters 1`] = `
<icon
  attr="default"
  attr2="my-attr"
  attr3="my attribute"
  className="my-class"
  id="my-id"
>
  
</icon>
`;

exports[`react block extensions should add an extension with an attribute 1`] = `
<icon
  attr="my-attr"
>
  
</icon>
`;

exports[`react block extensions should add an extension with an id 1`] = `
<icon
  id="my-id"
>
  
</icon>
`;

exports[`react block extensions should work with other markdown 1`] = `
Array [
  <h1>
    heading
  </h1>,
  "
",
  <icon>
    
  </icon>,
]
`;

exports[`react inline extensions should add an extension 1`] = `
<p>
  <icon />
</p>
`;

exports[`react inline extensions should add an extension with a boolean property 1`] = `
<p>
  <icon
    highlight={true}
  />
</p>
`;

exports[`react inline extensions should add an extension with a class 1`] = `
<p>
  <icon
    className="my-class"
  />
</p>
`;

exports[`react inline extensions should add an extension with a lone attribute 1`] = `
<p>
  <icon
    attr=""
  />
</p>
`;

exports[`react inline extensions should add an extension with a lone attribute with a default boolean value 1`] = `
<p>
  <icon
    highlight={true}
  />
</p>
`;

exports[`react inline extensions should add an extension with a lone attribute with a default value 1`] = `
<p>
  <icon
    attr="default"
  />
</p>
`;

exports[`react inline extensions should add an extension with a mapped argument 1`] = `
<p>
  <icon
    icon="my-icon"
  />
</p>
`;

exports[`react inline extensions should add an extension with a mapped content 1`] = `
<p>
  <icon
    tooltip="my-tooltip"
  />
</p>
`;

exports[`react inline extensions should add an extension with a quoted attribute 1`] = `
<p>
  <icon
    attr="my attribute"
  />
</p>
`;

exports[`react inline extensions should add an extension with all types of parameters 1`] = `
<p>
  <icon
    attr="default"
    attr2="my-attr"
    attr3="my attribute"
    className="my-class"
    id="my-id"
  />
</p>
`;

exports[`react inline extensions should add an extension with an attribute 1`] = `
<p>
  <icon
    attr="my-attr"
  />
</p>
`;

exports[`react inline extensions should add an extension with an id 1`] = `
<p>
  <icon
    id="my-id"
  />
</p>
`;

exports[`react inline extensions should work with other markdown 1`] = `
Array [
  <h1>
    heading
  </h1>,
  "
",
  <p>
    <icon />
  </p>,
]
`;
